### DISCUSSION TITLE
Sending encrypted emails with PHPMailer

### DISCUSSION THREAD

**Discussion by GalacticSun** [PHPMailer/PHPMailer#2751] [Q&A] [answered]
Hi there,

How do I go about sending encrypted emails with PHPMailer?

Is it even possible?

Thanks.

**Comment by Synchro** [comment, marked as answer]
There isn't a supported method for sending encrypted messages, but you can sign messages cryptographically with S/MIME.

Quite a while ago someone did post an example snippet that did encryption too, but it's never been wrapped up into a feature. A search in here should find it. It's also really tricky to make it all work without breaking DKIM.

**Reply by Darkovibe** [reply]
Hello I’m new here. Still don’t know how it all works!!

**Comment by GalacticSun** [comment]
Okay, thanks for letting me know.  "Signing" email is a way to verify that it is from the actual person who sent it, correct?  Whereas encryption is different, in that it encrypts the entire email message, requiring decryption on the receiving end?

**Reply by Synchro** [reply]
Yes, that's right. There was some discussion of this [way back in 2013](https://github.com/PHPMailer/PHPMailer/issues/127), but it never resulted in a merge.

**Reply by GalacticSun** [reply]
Interesting.  Is DKIM the same as (or a form of) signing email?

That PR never made it in, I guess?  Did that code that ratatine provided break DKIM or was it okay?

**Reply by Synchro** [reply]
Yes, DKIM is a kind of externalised signing that stores public keys in DNS, allowing receivers to verify that the sender is authentic and that a message has not been tampered with in transit. In some ways it's similar to S/MIME signing, the main difference is key distribution and how receivers handle it. Also DKIM generally implemented on mail servers rather than clients, though clients can do it too. DKIM uptake has vastly outstripped S/MIME signing, and verification is implemented by all major email providers (Yahoo invented it). The issue is that DKIM signing is *extremely* sensitive to tiny changes in message content, and PGP and S/MIME encryption make very large changes, so it's tricky to make DKIM still work, especially given that PHPMailer's internals are a little inconsistent (being generous here!). If you really need it, I'd recommend looking at Laminas mail or Symfony's mail component (used to be SwiftMailer), which you should be able to use without having to use their whole frameworks, and I'm fairly sure they both offer S/MIME encryption.

**Reply by GalacticSun** [reply]
Thanks a lot, I appreciate you taking the time and explaining this for me.  PHPMailer is great and thanks for creating it.  All the best.
